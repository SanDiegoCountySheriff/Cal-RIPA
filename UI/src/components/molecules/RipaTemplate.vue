<template>
  <v-card class="mx-auto" max-width="500" flat>
    <v-card-text>
      <div class="tw-mt-4 tw-mb-4">
        <v-container fluid>
          <v-row no-gutters dense>
            <v-col cols="12" sm="12" class="tw-mt-4 tw-text-center">
              <v-btn
                :disabled="disableButtons"
                color="primary"
                @click="handleDefaultTemplate"
              >
                BASIC STOP
              </v-btn>
            </v-col>
          </v-row>

          <v-row>
            <v-col
              cols="12"
              sm="6"
              class="tw-text-center"
              v-for="(template, index) in stopTemplates"
              :key="index"
            >
              <v-btn
                :disabled="disableButtons"
                outlined
                color="primary"
                @click="handleDynamicTemplates(template.displayName)"
              >
                {{ template.displayName }}
              </v-btn>
            </v-col>
          </v-row>
        </v-container>
      </div>

      <v-divider></v-divider>

      <div class="tw-mt-4">About this App:</div>
      <p>
        The Racial and Identity Profiling Act of 2015 (AB 953) requires state
        and local law enforcement agencies, to collect data regarding stops of
        individuals, including perceived demographic information on the person
        stopped, and to report this data to the California Attorney General's
        (AG) Office.
      </p>
      <p>
        The AGâ€™s Office has adopted
        <a
          href="https://oag.ca.gov/sites/all/files/agweb/pdfs/ripa/stop-data-reg-final-text-110717.pdf"
          target="_blank"
          rel="noopener noreferrer"
        >
          these regulations
        </a>
        on November 7, 2017. For more information please see the
        <a
          href="https://oag.ca.gov/ab953/regulations"
          target="_blank"
          rel="noopener noreferrer"
        >
          AG's Website </a
        >.
      </p>
      <v-divider></v-divider>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  name: 'ripa-template',

  methods: {
    handleDynamicTemplates(name) {
      if (this.onOpenTemplate) {
        this.onOpenTemplate(name)
      }
    },

    handleDefaultTemplate() {
      if (this.onOpenTemplate) {
        this.onOpenTemplate()
      }
    },
  },

  props: {
    stopTemplates: {
      type: Array,
      default: () => [],
    },
    onOpenTemplate: {
      type: Function,
      required: true,
    },
    disableButtons: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
