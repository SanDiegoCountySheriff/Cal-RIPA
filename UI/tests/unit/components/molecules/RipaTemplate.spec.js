import RipaTemplate from '@/components/molecules/RipaTemplate.vue'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Template', () => {
  let wrapper
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return shallowMount(RipaTemplate, {
      vuetify,
      propsData: {
        ...propsData,
        onOpenTemplate: jest.fn(),
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaTemplate, {
      vuetify,
      propsData: {
        onOpenTemplate: jest.fn(),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it.todo('should handle dynamic templates')

  it.todo('should handle default template')
})
