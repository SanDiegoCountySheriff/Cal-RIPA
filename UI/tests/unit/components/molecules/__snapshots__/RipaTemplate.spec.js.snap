// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Template should match snapshot 1`] = `
<div class="mx-auto v-card v-card--flat v-sheet theme--light" style="max-width: 500px;">
  <div class="v-card__text">
    <div class="tw-mt-4 tw-mb-4">
      <div class="container container--fluid">
        <div class="row no-gutters row--dense">
          <div class="tw-mt-4 tw-text-center col-sm-12 col-12"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">
              BASIC STOP
            </span></button></div>
        </div>
        <div class="row"></div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="tw-mt-4">About this App:</div>
    <p>
      The Racial and Identity Profiling Act of 2015 (AB 953) requires state
      and local law enforcement agencies, to collect data regarding stops of
      individuals, including perceived demographic information on the person
      stopped, and to report this data to the California Attorney General's
      (AG) Office.
    </p>
    <p>
      The AGâ€™s Office has adopted
      <a href="https://oag.ca.gov/sites/all/files/agweb/pdfs/ripa/stop-data-reg-final-text-110717.pdf" target="_blank" rel="noopener noreferrer">
        these regulations
      </a>
      on November 7, 2017. For more information please see the
      <a href="https://oag.ca.gov/ab953/regulations" target="_blank" rel="noopener noreferrer">
        AG's Website </a>.
    </p>
    <p>To report any bugs or request features please
      <a href="https://github.com/SanDiegoCountySheriff/Cal-RIPA/issues" target="_blank" rel="noopener noreferrer">submit an issue</a>.
    </p>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  </div>
</div>
`;
